packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -421,21 +421,23 @@ export class HttpPlugin extends BasePlugin<Http> {
       const spanOptions: SpanOptions = {
         kind: SpanKind.CLIENT,
       };
-      const span = plugin._startHttpSpan(operationName, spanOptions);
-
-      return plugin._tracer.withSpan(span, () => {
-        if (!optionsParsed.headers) optionsParsed.headers = {};
-        propagation.inject(optionsParsed.headers);
+      const headers = optionsParsed.headers || {};
+      return context.with(propagation.extract(headers), () => {
```

Comment:
I am surprised to see a call to `extract` while making an outgoing http request. Typically this will be done when receiving an inbound request. What context are you extracting and how was it injected into the outbound headers in the first place?

Reply:
The span is being created for outgoing request, this span has no context even though the propagation from headers is available. This is especially visible when using opentracing and you have only headers from B3 for example. Otherwise the newly created span has no context and a new will be created, so then you have 2 groups of spans instead of 1

Reply:
When we extract the context from this header, where is it injected from? Does that injection happen in `utils.getRequestInfo`? I agree with @mwear this seems like an odd pattern being used to cover up an underlying context propagation problem. Is there a simple example where you can show that the context is lost without this change?

Reply:
Guys this is not forcing a new context etc. This only sets a context based on injected headers, you have the same logic in _incomingRequestFunction line 313. So without this the context is lost
The simple example is the unit test I added - explains exactly what is happening and failing without this fix.
There is also more advanced example with opentracing and client - server if you interested -> https://github.com/lightstep/opentelemetry-examples/tree/main/nodejs/ot-shim and simply check the main screennshot. Without this you will have 2 groups of spans and can't really see the whole request from client -server- client

Reply:
Hmm. To me it is obvious that you have to extract context on an incoming request (_incomingRequestFunction line 313). It is not at all obvious to me why this would be done for an outgoing request. I'm not confused on what is happening, I think the test explains that quite well. What I am not understanding is why you would need it.

Reply:
Because otherwise the created span is not assigned to corrected context and instead of being able to see the whole flow from client sending a request to server and server responds with message back to the client. This was missing as you extract the context from headers anyway and this context is not then propagated to any new created span so this is obvious bug.

Reply:
our instrumentation is creating a span for outgoing request and this span has wrong context not the one that should be propagated.

Reply:
> Because otherwise the created span is not assigned to corrected context

I don't think i agree too, the correct context is the one that launched the request, currently this should be the HTTP incoming patch but it might change if the end user have custom span (in which case the outgoing span should be the child of the custom span). With your PR, you would get whatever context was injected in the headers prior to the request being sent (and i think currently nothing is injected there anyway).

Reply:
> > Because otherwise the created span is not assigned to corrected context
> 
> I don't think i agree too, the correct context is the one that launched the request, currently this should be the HTTP incoming patch but it might change if the end user have custom span (in which case the outgoing span should be the child of the custom span). With your PR, you would get whatever context was injected in the headers prior to the request being sent (and i think currently nothing is injected there anyway).

the thing is that context is empty completely - so a new context is being created when new span is created, there is no propagation of anything, not a context which you mentioned about, nothing is propagated

Reply:
I would encourage you guys to run the code, from the example I posted (https://github.com/lightstep/opentelemetry-examples/tree/main/nodejs/ot-shim) and then debug the line in outgoing request with my change and without and see that you will have 2 groups of not related spans to each other and then try my fix and see the same

Reply:
I have just read carefully all the discussion that went here -> https://github.com/open-telemetry/opentelemetry-js/issues/1479 and this is exactly the problem I'm fixing here too it is has just different use case.
The new span is then created in the context of previous span, whereas if the headers has information about context it should not.
Moreover just copied one unit test from this PR from @vmarchaud  https://github.com/open-telemetry/opentelemetry-js/pull/1546/files
And the test passes too.
```
it('should not set span as active in context for outgoing request', async () => {
        assert.deepStrictEqual(getActiveSpan(context.active()), undefined);
        await httpRequest.get(`${protocol}://${hostname}:${serverPort}/test`);
        assert.deepStrictEqual(getActiveSpan(context.active()), undefined);
        const spans = memoryExporter.getFinishedSpans();
        assert.strictEqual(spans.length, 2);
      });
```

Reply:
The test in the PR as mentionned [here](https://github.com/open-telemetry/opentelemetry-js/pull/1546#discussion_r491867943) doesn't actually test the logic.

> I have just read carefully all the discussion that went here -> #1479 and this is exactly the problem I'm fixing here too it is has just different use case.

Well in the case of the issue, the context isn't empty (it contains the http server span) so i don't see why you keep saying that.

I'm curious how this will fix the issue anyway since you are still setting the span for the outgoing request as active in the context too.

I'll re-open my PR for others to choose how to handle that.